### PR TITLE
Sphinx docs2

### LINKED ISSUE #306: Try out building documentation with Sphynx and export it to LaTeX
It would be great to have our Python documentation added to our ArchDoc as an appendix. The documentation building tool Sphynx can do this, but we don't know how difficult it is to set up.

### PR DESCRIPTION
**Pull request by Qqwy**
- Adds Sphinx to create documentation of our modules
- Adds 'documentation' task to `build_and_run.sh` that runs Sphinx in such a way that it creates a PDF.

Not in this PR:
- General documentation (which might be added to `pydash/sphinx_docs/index.rst`
- Fixing style of document to be able to add it as an appendix to our other documents.
- Some problems in the docstrings that result in oddly looking LaTeX (important for #443 )

Fixes #306

### CODE REVIEW

**Comment by Qqwy** [comment]
Fixes #306

**Comment by InputUsername** [comment]
Nice! :+1:

Two remarks:
- Some documentation doesn't seem to parse correctly (see image below)
- Maybe export to the `docs` directory instead of `pydash`

Example image (taken from the exported PDF):
image

**Comment by Qqwy** [comment]
- remark (1): Sphinx expects docstrings to be written in RST format.
- remark (2): Unfortunately, there is no easy way to do this AFAIK. I might be able to copy the finished PDF over to the `docs` folder, however.

But neither of these two remarks should block this PR.

**Comment by Qqwy** [comment]
No idea why Travis does not like this branch, by the way

File: pydash-front/src/authenticated_app/dashboard/statistics_page/StatisticsPage.js.orig [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+                    }
+                });
+            } else {
+                this.setState(prevState => {
+                    return {
+                        ...prevState,
+
+
+                        dashboard: response.data,
+<<<<<<< HEAD
```
Comment (InputUsername) [inline pydash-front/src/authenticated_app/dashboard/statistics_page/StatisticsPage.js.orig]: still some artefacts of a conflict here

File: pydash-front/src/authenticated_app/dashboard/statistics_page/StatisticsPage.js.orig [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+        }).then((response) => {
+            let total_visits = (response.data && response.data.aggregates && response.data.aggregates.total_visits) || 0;
+            console.log("Response: ", response);
+            if (response.data.hasOwnProperty('error')) {
+                this.setState(prevState => {
+                    return {
+                        ...prevState,
+
+                        dashboard: response.data,
+<<<<<<< HEAD
```
Comment (InputUsername) [inline pydash-front/src/authenticated_app/dashboard/statistics_page/StatisticsPage.js.orig]: conflict

File: pydash-front/src/authenticated_app/endpoint/EndpointsTable.js.orig [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    handleType = key => event => {
+        this.setState({
+            [key]: event.target.value
+        });
+      };
+
+    render = () => {
+        console.log("ENDPOINTS TABLE", this.props.data);
+<<<<<<< HEAD
```
Comment (InputUsername) [inline pydash-front/src/authenticated_app/endpoint/EndpointsTable.js.orig]: conflict

File: pydash-front/src/authenticated_app/dashboard/statistics_page/StatisticsPage.js.orig (hunk 1 above)
Comment (InputUsername) [reply pydash-front/src/authenticated_app/dashboard/statistics_page/StatisticsPage.js.orig]: wait, what are these `.orig` files for?

Comment (Qqwy) [reply pydash-front/src/authenticated_app/dashboard/statistics_page/StatisticsPage.js.orig]: They are an artefact that Git creates when doing conflict resolution on the command line. I've removed these files.

**Pull request merged**